Surface_mesh_simplification/include/CGAL/Surface_mesh_simplification/Policies/Edge_collapse/FastEnvelope_filter.h

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2020  GeometryFactory (France). All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Andreas Fabri
+//
+#ifndef CGAL_SURFACE_MESH_SIMPLIFICATION_POLICIES_EDGE_COLLAPSE_FAST_ENVELOPE_FILTER_H
+#define CGAL_SURFACE_MESH_SIMPLIFICATION_POLICIES_EDGE_COLLAPSE_FAST_ENVELOPE_FILTER_H
+
+#include <CGAL/license/Surface_mesh_simplification.h>
+
+#include <CGAL/assertions.h>
+#include <CGAL/Default.h>
+#include <CGAL/intersections.h>
+#include <CGAL/boost/graph/named_params_helper.h>
+#include<CGAL/Surface_mesh_simplification/internal/common.h>
+
+#include <fastenvelope/FastEnvelope.h>
+#include <fastenvelope/Types.hpp>
+
+#include <boost/optional.hpp>
+
+#include <vector>
+#include <type_traits>
+
+namespace CGAL {
+namespace Surface_mesh_simplification {
+
+
+template<typename GeomTraits,typename BaseFilter = internal::Dummy_filter>
+class FastEnvelope_filter
```

Comment:
doc says `Envelope_filter`

Reply:
`Polyhedral_envelope_filter` would be better IMHO

Reply:
This is the wrapper around the original FastEnvelope code.  Shall we just remove it?

Reply:
As I told you I was reluctant to change _envelope_ to _polyhedral envelope_ in the class names as other reviewers might come up with an even better name, but as it is now consistently used in the User Manual I should probably just rename it.

Reply:
After a discussion with @sloriot we decided to consistently replace `Envelope` with `Polyhedral_envelope`.